messages/teleporter/message_manager.go

```diff
@@ -90,6 +90,20 @@ func isAllowedRelayer(allowedRelayers []common.Address, eoa common.Address) bool
 	return false
 }
 
+func (m *messageManager) GetDestinationChainID(warpMessageInfo *vmtypes.WarpMessageInfo) (ids.ID, error) {
+	// Unpack the teleporter message to get the destination chain ID
+	teleporterMessage, err := teleportermessenger.UnpackTeleporterMessage(warpMessageInfo.WarpPayload)
```

Comment:
Can you put the unpacked Teleporter message into `teleporterMessageCache`, and update `ShouldSendMessage` to check the cache first? That way we don't need to unpack the message in both methods.

Reply:
fixed